### ISSUE TITLE
Pub package manager does not handle native extensions

### ISSUE THREAD

**Issue by DartBot** [dart-lang/pub#250]
<a href="https://github.com/whesse"><img src="https://avatars.githubusercontent.com/u/4905639?v=3" align="left" width="96" height="96"hspace="10"></img></a> **Issue by [whesse](https://github.com/whesse)**
_Originally opened as dart-lang/sdk#6290_

----

I don't think that pub will correctly manage and download the native extension shared libraries for a Dart extension.  These are typically dart libraries with an import statement like

#import(&quot;dart-ext:foo&quot;),

which makes the standalone Dart binary load a library named foo.dll, libfoo.so, or libfoo.dylib, and run the initialization routine.  To distribute these packages, pub would have to download the appropriate native library to the package cache.

This is also blocked on proper handling of the package: prefix in the URL of the importing library, when figuring out the path of the native shared library from the (relative) path in the dart-ext: import.  This is issue dart-lang/sdk#6264.

**Comment by DartBot** [comment]
<a href="https://github.com/mezoni"><img src="https://avatars.githubusercontent.com/u/2311549?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [mezoni](https://github.com/mezoni)**

----

Once again you are part approach to solving this problem.
I'll give a little hint.

At what host machine they will be compiled and linked?
Not all operating systems are free.

Not everyone can be on your machine have a different number of platforms.

I think that soon you will not solve this problem.

Unless, of course, will not be found any suitable solution. Compromise.

Or do you have a deck of cards with a more than four aces.

**Comment by DartBot** [comment]
<a href="https://github.com/sethladd"><img src="https://avatars.githubusercontent.com/u/5479?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [sethladd](https://github.com/sethladd)**

----

*Added [Duplicate](../labels/Duplicate) label.*
*Marked as being merged into dart-lang/pub#39.*
